solidserver/data_source_ip_address.go

```diff
@@ -0,0 +1,146 @@
+package solidserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math"
+	"net/url"
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceipaddress() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceipaddressRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeInt,
+				Description: "The id of the ip address",
+				Required:    true,
+			},
+			"space": {
+				Type:        schema.TypeString,
+				Description: "The name of the space into which creating the IP address.",
+				Computed:    true,
+			},
+			"subnet": {
+				Type:        schema.TypeString,
+				Description: "The name of the subnet into which creating the IP address.",
+				Computed:    true,
+			},
+			"pool": {
+				Type:        schema.TypeString,
+				Description: "The name of the pool into which creating the IP address.",
+				Computed:    true,
+			},
+			"address": {
+				Type:        schema.TypeString,
+				Description: "The provisionned IP address.",
+				Computed:    true,
+			},
+			"device": {
+				Type:        schema.TypeString,
+				Description: "Device Name to associate with the IP address (Require a 'Device Manager' license).",
+				Computed:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The long name or FQDN of the IP address to create.",
+				Computed:    true,
+			},
+			"mac": {
+				Type:        schema.TypeString,
+				Description: "The MAC Address of the IP address to create.",
+				Computed:    true,
+			},
+
+			"class": {
+				Type:        schema.TypeString,
+				Description: "The class associated to the IP address.",
+				Computed:    true,
+			},
+			"cidr": {
```

Comment:
By CIDR, do you mean the prefix-size ? If so, I would suggest to name this field "prefix-size" as I plan to to add both the netmask and prefix (CIDR notation) to the data-source and resource.

Reply:
right, and maybe call it "prefix-length" to be precise ?